### PR TITLE
RDM-9580

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
RDM-9580: Database Query Timeouts to be Revisited
    
           [RDM-9580] (https://tools.hmcts.net/jira/browse/RDM-9580).

### CODE REVIEW

File: application/src/main/java/uk/gov/hmcts/ccd/definition/store/ApplicationConfiguration.java [hunk 1]
```diff
@@ -40,4 +44,12 @@ public Executor asyncExecutor(@Value("${async.executor.core.pool.size}") Integer
     public CacheManagerCustomizer<CaffeineCacheManager> cacheManagerCustomizer() {
         return cacheManager -> cacheManager.setAllowNullValues(false);
     }
+
+    @Bean
+    public PlatformTransactionManager transactionManager(final EntityManagerFactory emf,@Value("${ccd.tx-timeouts.default}") String timeout) {
+        final CustomTransactionManager transactionManager = new CustomTransactionManager();
+        transactionManager.setEntityManagerFactory(emf);
+        transactionManager.configureTxTimeout(ImportController.class, "processUpload", Integer.parseInt(timeout));
```
Comment (Thor-tech-of-metal) [inline application/src/main/java/uk/gov/hmcts/ccd/definition/store/ApplicationConfiguration.java:52]: Here is the part where we setup which transaction  has the defined timeout.

File: application/src/main/java/uk/gov/hmcts/ccd/definition/store/ApplicationConfiguration.java [hunk 2]
```diff
@@ -40,4 +44,12 @@ public Executor asyncExecutor(@Value("${async.executor.core.pool.size}") Integer
     public CacheManagerCustomizer<CaffeineCacheManager> cacheManagerCustomizer() {
         return cacheManager -> cacheManager.setAllowNullValues(false);
     }
+
+    @Bean
+    public PlatformTransactionManager transactionManager(final EntityManagerFactory emf,@Value("${ccd.tx-timeouts.default}") String timeout) {
```
Comment (rajaram-nagarajan) [inline application/src/main/java/uk/gov/hmcts/ccd/definition/store/ApplicationConfiguration.java:49]: Could we move this 'timeout' from the parameter to inside the method?

File: excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/endpoint/ImportController.java [hunk 3]
```diff
@@ -50,9 +52,17 @@ public ImportController(ImportServiceImpl importService,
         this.azureStorageConfiguration = azureStorageConfiguration;
     }
 
-    @Transactional
+    private void print(String message){
```
[the remark this answers is not in the document]
Comment (Thor-tech-of-metal) [reply excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/endpoint/ImportController.java:55]: Yes !

File: excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/endpoint/ImportController.java [hunk 4]
```diff
@@ -78,11 +88,12 @@ public ResponseEntity processUpload(@RequestParam("file") MultipartFile file) th
...
                     LOG.warn(warning);
                 }
+                print("Ending:");
                 return ResponseEntity.status(HttpStatus.CREATED)
                     .header(IMPORT_WARNINGS_HEADER, importService.getImportWarnings().toArray(new String[0]))
                     .body(responseBody);
             }
-
+            print("Ending:");
             return ResponseEntity.status(HttpStatus.CREATED).body(responseBody);
```
Comment (Thor-tech-of-metal) [inline excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/endpoint/ImportController.java:97]: That one needs to be removed too.

File: application/src/main/java/uk/gov/hmcts/ccd/definition/store/ApplicationConfiguration.java (hunk 2 above)
Comment (Thor-tech-of-metal) [reply application/src/main/java/uk/gov/hmcts/ccd/definition/store/ApplicationConfiguration.java:49]: this is not a common attribute to the hole class, timeout is not an attribute .  It is a parameter only used for Bean creation.

File: charts/ccd-definition-store-api/values.yaml [hunk 5]
```diff
@@ -20,7 +20,7 @@ java:
     DEFINITION_STORE_DB_USERNAME: ccd@ccd-definition-store-api-postgres-db-{{ .Values.global.environment }}
     DEFINITION_STORE_DB_OPTIONS: "?sslmode=require"
     DEFINITION_STORE_DB_MAX_POOL_SIZE: 32
-
+    DEFINITION_TX_TIMEOUTS_DEFAULT: 20
```
Comment (Thor-tech-of-metal) [inline charts/ccd-definition-store-api/values.yaml:24]: Please evaluate if this is the correct value for prod , 20 seconds

File: elastic-search-support/src/main/resources/application.yml [hunk 6]
```diff
@@ -2,10 +2,10 @@ elasticsearch:
   host: ${ELASTIC_SEARCH_HOST:localhost}
   port: ${ELASTIC_SEARCH_PORT:9200}
   scheme: ${ELASTIC_SEARCH_SCHEME:http}
-  enabled: ${ELASTIC_SEARCH_ENABLED:false}
```
[the remark this answers is not in the document]
Comment (Thor-tech-of-metal) [reply elastic-search-support/src/main/resources/application.yml:5]: please revert it . That should not be in the PR

**Review by MSancaktutar — APPROVED** [review]
LGTM

**Pull request merged**